### PR TITLE
Serialization paths unification

### PR DESCRIPTION
**Pull request by atobiszei**
Additionally fixed passing arguments to GMock framework in unit tests.

JIRA:CVS-59331

### CODE REVIEW

**Comment by atobiszei** [comment]
TODO to remove:
https://github.com/openvinotoolkit/model_server/pull/762/files#diff-45387e9b34b1f44657d9695c662ca85f62c862314ef532ab493d6d2812aa776bR48

File: src/serialization.cpp [hunk 1]
```diff
@@ -62,36 +62,61 @@ Status serializeBlobToTensorProto(
     }
     }
     responseOutput.mutable_tensor_shape()->Clear();
-    for (auto dim : networkOutput->getEffectiveShape()) {
+    auto& effectiveNetworkOutputShape = networkOutput->getEffectiveShape();
+    auto& actualBlobShape = blob->getTensorDesc().getDims();
+    for (size_t i = 0; i < effectiveNetworkOutputShape.size(); ++i) {
+        size_t dim;
+        if (effectiveNetworkOutputShape[i] != 0) {
+            dim = effectiveNetworkOutputShape[i];
```
Comment (dkalinowski) [inline src/serialization.cpp:70]: We should validate blob precision and shape against tensorinfo precision and shape

File: src/test/ensemble_tests.cpp [hunk 2]
```diff
@@ -664,7 +722,12 @@ TEST_F(EnsembleFlowTest, FailInDLNodeFetchResultsStreamIdReleasedForDeferredNode
     auto input_node = std::make_unique<EntryNode>(&request, inputsInfo);
     auto failInFetchNode = std::make_unique<DLNodeFailInFetch>("failInFetch_node", dummyModelName, requestedModelVersion, managerWithDummyModel);
     auto modelNode = std::make_unique<DLNodeFailInFetch>("dummy_node", dummyModelName, requestedModelVersion, managerWithDummyModel);
-    auto output_node = std::make_unique<ExitNode>(&response);
+    auto tensorInfo = std::make_shared<ovms::TensorInfo>(customPipelineOutputName,
```
Comment (dkalinowski) [inline src/test/ensemble_tests.cpp:725]: Maybe we could separate this tensorInfo to test_utils or fixture field? This way we would not need to initialize it in each test.

File: src/test/unit_tests.cpp [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 int main(int argc, char** argv) {
     ::testing::InitGoogleTest(&argc, argv);
+    ::testing::InitGoogleMock(&argc, argv);
```
Comment (dkalinowski) [inline src/test/unit_tests.cpp:21]: Why wasn't it required before?

Comment (atobiszei) [reply src/test/unit_tests.cpp:21]: It just enables you to use those arguments. In my case it was using `--gmock_verbose=info`.

Without the Init it was just not passed into GMock. If you won't do that then nothing happens.

**Review by michalkulakowski — APPROVED** [review]

**Comment by michalkulakowski** [comment]
LGTM

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**